File: mixin/src/main/java/com/idkidknow/mcreallink/mixin/complement/BroadcastingMessage.kt
package com.idkidknow.mcreallink.mixin.complement

import net.minecraft.network.chat.Component

/** @see com.idkidknow.mcreallink.mixin.mixin.PlayerListMixin */
object BroadcastingMessage {
    private val _ignoreMessage = ThreadLocal.withInitial { false }
    val ignoreMessage: Boolean get() = _ignoreMessage.get()
    var callback: (Component) -> Unit = {}

    fun ignoreTemporarily(action: () -> Unit) {
        _ignoreMessage.set(true)
        action()
        _ignoreMessage.set(false)
    }
}

File: versions/forge1182/src/main/scala/com/idkidknow/mcreallink/forge1182/mixin/complement/BroadcastingMessage.scala
package com.idkidknow.mcreallink.forge1182.mixin.complement

import net.minecraft.network.chat.Component

object BroadcastingMessage {
  private val _ignoreMessage = ThreadLocal.withInitial(() => false)
  def ignoreMessage: Boolean = _ignoreMessage.get()
  @SuppressWarnings(Array("org.wartremover.warts.Var"))
  var callback: Component => Unit = { _ => () }

  def ignoreTemporarily(action: () => Unit): Unit = {
    _ignoreMessage.set(true)
    action()
    _ignoreMessage.set(false)
  }
}
